business/namespaces.go

```diff
@@ -129,12 +129,98 @@ func (in *NamespaceService) GetNamespaces(ctx context.Context) ([]models.Namespa
 		labelSelectorExcludeValue = excludeLabelList[1]
 	}
 
+	namespaces := []models.Namespace{}
+
+	var clusterNames []string
+	for c := range in.userClients {
+		clusterNames = append(clusterNames, c)
+	}
+
+	wg := &sync.WaitGroup{}
+	type result struct {
+		cluster string
+		ns      []models.Namespace
+		err     error
+	}
+	resultsCh := make(chan result)
+
+	go func() {
+		for _, cluster := range clusterNames {
+			wg.Add(1)
+			go func(c string) {
+				defer wg.Done()
+				list, error := in.GetNamespacesByCluster(c)
+				if error != nil {
+					log.Errorf("Error fetching Namespaces per cluster %s: %s", c, error)
+					resultsCh <- result{cluster: c, ns: nil, err: error}
+				} else {
+					resultsCh <- result{cluster: c, ns: list, err: nil}
+				}
+			}(cluster)
+		}
+		wg.Wait()
+		close(resultsCh)
+	}()
+
+	// Combine namespace data
+	for result := range resultsCh {
+		if result.err != nil {
+			log.Errorf("Error fetching Namespaces per cluster %s: %s", result.cluster, result.err)
```

Comment:
the user may not have permission to see the cluster right? In that case this should not be an error. At most it should be a warning (but even then I'm not sure - maybe this should be Info or even Debug?)

Reply:
Right. We can shown an error when it is from the home cluster. For the remote cluster, maybe we can use an info: It could be more important as this cannot be just from permissions issue, maybe a network issue, or other reason - But yes, there could be many messages.